From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: nostalfinals <yuu8583@proton.me>
Date: Sat, 15 Jun 2024 11:02:45 +0800
Subject: [PATCH] Publish API


diff --git a/build.gradle.kts b/build.gradle.kts
index ea3ad97909de951efa0ad322df766689339b8af6..4b17e1651c6c4885ba13e7a4390135985235e169 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -223,3 +223,14 @@ tasks.check {
     dependsOn(scanJarForOldGeneratedCode)
 }
 // Paper end
+
+// charon start - publish api
+publishing {
+    repositories {
+        maven("https://maven.nostal.ink/repository/maven-snapshots/") {
+            name = "nostal"
+            credentials(PasswordCredentials::class)
+        }
+    }
+}
+// charon end - publish api
\ No newline at end of file
